src/todolist/CommandParserImpl.java

```diff
@@ -0,0 +1,25 @@
+package todolist;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+public class CommandParserImpl implements CommandParser {
+
+    @Override
+    public ParseResult parse(String source) {
+        String[] sourses = source.trim().split(" ");
```

Comment:
Нужно проверить, что source != null
```suggestion
        if(source == null) {
          return new ParseResult.Error("Source should not be null");
        }
        String[] sourses = source.trim().split(" ");
```